### DISCUSSION TITLE
How do you use addUpgradeStatement?

### DISCUSSION THREAD

**Discussion by khromov** [capacitor-community/sqlite#482] [Q&A] [answered]
I'm trying to find a an example of applying multiple upgrade statements but can't find any working examples. I have something like this right now, but it does not work, only `version1` is applied, and `version2` is completely ignored. How do you apply multiple upgrade statements in a row? An example would be amazing.

```ts
const DB_NAME = 'my-db';

const version1 = [
	`CREATE TABLE "journal_entries" (
		"id"	INTEGER UNIQUE,
		"date"	INTEGER NOT NULL,
		"text"	TEXT,
		"attachment"	BLOB,
		PRIMARY KEY("id" AUTOINCREMENT)
	);`
];

const version2 = [
	`ALTER TABLE "journal_entries" ADD COLUMN "attachment_2" BLOB;`
];

await sqlite.addUpgradeStatement(DB_NAME, 1, version1);
await sqlite.addUpgradeStatement(DB_NAME, 2, version2);
// ...etc
	
const db = await sqlite?.createConnection(...}
```

**Comment by khromov** [comment]
I think I figured it out, you have to specify the DB version in the `createConnection` statement, eg:

```ts
const db = await sqlite?.createConnection(
				DB_NAME,
				false,
				'no-encryption',
				2, // db version here
				false
			);
```

One thing that I do not understand is how the sqlite plugin keeps track of which version the database is currently at. Is it safe to assume that adding upgrade statements and incrementing the version in `createConnection` is enough for a successful migration regardless of how which version the database is currently at or are there some other considerations?

**Comment by jepiqueau** [comment, marked as answer]
@khromov look at [tutorials](https://jepiqueau.github.io/)
